packages/backend/src/services/warehouseClients/DatabricksWarehouseClient.ts

```diff
@@ -0,0 +1,109 @@
+import odbc, { Result } from 'odbc';
+import { CreateDatabricksCredentials, DimensionType } from 'common';
+import { WarehouseCatalog, WarehouseClient } from '../../types';
+import { WarehouseConnectionError, WarehouseQueryError } from '../../errors';
+
+export const DRIVER_PATH = '/opt/simba/spark/lib/64/libsparkodbc_sb64.so';
+
+type SparkSchemaResult = {
+    TABLE_CAT: string;
+    TABLE_SCHEM: string;
+    TABLE_NAME: string;
+    COLUMN_NAME: string;
+    DATA_TYPE: number;
+    TYPE_NAME: string;
+};
+
+export default class DatabricksWarehouseClient implements WarehouseClient {
+    connectionString: string;
+
+    constructor({
+        serverHostName,
+        port,
+        personalAccessToken,
+        httpPath,
+    }: CreateDatabricksCredentials) {
+        this.connectionString = `Driver=${DRIVER_PATH};Server=${serverHostName};HOST=${serverHostName};PORT=${port};SparkServerType=3;Schema=default;ThriftTransport=2;SSL=1;AuthMech=3;UID=token;PWD=${personalAccessToken};HTTPPath=${httpPath};UseNativeQuery=1`;
```

Comment:
Do we need to have the ssl mode as an input in the UI ? 
We can implement that later if this is the common setup.

Reply:
Yep this seems to be standard and always the case for Databricks. Probably for different spark clusters it would change.